File: snippets/csharp/xml-production-logging_main.cs
using Serilog;
using Microsoft.Agents.Protocol.Client;

public class ProductionAgentService
{
    private readonly AgentProtocolClient _client;
    private readonly ILogger _logger;

    public ProductionAgentService(string endpoint, ILogger logger)
    {
        // Enable automatic logging to files
        _client = new AgentProtocolClient(
            endpoint,
            enableLogging: true,
            logDirectory: "logs/production"
        );
        _logger = logger;
    }

    public async Task<string> ChatAsync(string userInput, string? threadId = null)
    {
        var conversation = threadId == null
            ? _client.CreateConversation()
            : _client.ResumeConversation(threadId);

        var response = await conversation.SendAsync(userInput).ConfigureAwait(false);

        // Structured logging with message count
        _logger.Information(
            "Conversation turn completed: {ThreadId}, Messages: {Count}",
            conversation.ThreadId,
            conversation.Messages.Count);

        // Optionally stream XML to centralized storage
        await SendToObservabilityPlatform(conversation.ThreadId!, conversation.ToString()).ConfigureAwait(false);

        return response;
    }

    private async Task SendToObservabilityPlatform(string threadId, string xml)
    {
        // Send to Azure Monitor, Datadog, etc.
        await Task.CompletedTask.ConfigureAwait(false);
    }
}

File: snippets/python/xml-production-logging_main.py
import asyncio
import logging
from typing import Optional
from microsoft.agents.protocol import AgentProtocolClient


class ProductionAgentService:
    def __init__(self, endpoint: str):
        # Enable automatic logging to files
        self.client = AgentProtocolClient(
            endpoint,
            enable_logging=True,
            log_directory="logs/production"
        )
        self.logger = logging.getLogger(__name__)

    async def chat(self, user_input: str, thread_id: Optional[str] = None) -> str:
        if thread_id:
            conversation = self.client.resume_conversation(thread_id)
        else:
            conversation = self.client.create_conversation()

        response = await conversation.send(user_input)

        # Structured logging with message count
        self.logger.info(
            "Conversation turn completed",
            extra={
                "thread_id": conversation.thread_id,
                "message_count": len(conversation.messages)
            }
        )

        # Optionally stream XML to centralized storage
        await self.send_to_observability_platform(
            conversation.thread_id,
            str(conversation)
        )

        return response

    async def send_to_observability_platform(self, thread_id: str, xml: str) -> None:
        # Send to Datadog, New Relic, etc.
        pass


async def main() -> None:
    service = ProductionAgentService("http://localhost:5000")
    await service.chat("Hello, how can you help?")


if __name__ == "__main__":
    asyncio.run(main())
